### ISSUE TITLE
Redownlaod of corrupt chunks

### ISSUE THREAD

**Issue by SaculRennorb** [gpailler/MegaApiClient#54]
Every time a chunks checksum is not as expected the api will throw a DownloadException and therefore end the download. 
I know that this behavior is perfectly fine but something quite annoying when downlaoding big files since the downlaod has to be done all over again.

I also dont know if this is even possible due to the streaming donwload, but would it be possible to create an option for just "redownloading" the last chunk if the checksumm is incorrect?

**Comment by gpailler** [comment]
An option to re-download a chunk if checksum is incorrect is definitely possible but I never faced this issue in real life. What kind of connection do you have to face checksum issues?

**Comment by SaculRennorb** [comment]
well... I myself can't reproduce the error since my connection is quite good and stable, but for one of my programs, wich some of my friends use with way worse internet connection, this happens a lot. Since they  have downloads of up to a GB (or multiple of those). I can see the frustration when it downloads fine until 90%, then one package is corrupt and all was for nothing.

tl;dr: yea i never encountered it either, but my buddy just got relay sh*tty internet.

**Comment by gpailler** [comment]
Probably part of #40 and #44
